src/Data/String/NonEmpty/CodePoints.purs

```diff
@@ -33,19 +33,18 @@ import Data.Semigroup.Foldable (class Foldable1)
 import Data.Semigroup.Foldable as F1
 import Data.String.CodePoints (CodePoint)
 import Data.String.CodePoints as CP
-import Data.String.NonEmpty.Internal (NonEmptyString, fromString)
+import Data.String.NonEmpty.Internal (NonEmptyString(..), fromString)
 import Data.String.Pattern (Pattern)
 import Partial.Unsafe (unsafePartial)
-import Unsafe.Coerce (unsafeCoerce)
 
 toNonEmptyString :: String -> NonEmptyString
-toNonEmptyString = unsafeCoerce
+toNonEmptyString = NonEmptyString
 
 fromNonEmptyString :: NonEmptyString -> String
-fromNonEmptyString = unsafeCoerce
+fromNonEmptyString (NonEmptyString s) = s
 
 liftS :: forall r. (String -> r) -> NonEmptyString -> r
-liftS = unsafeCoerce
+liftS f (NonEmptyString s) = f s
```

Comment:
I know we don't typically do this, but is there value in adding comments along the lines of "For internal use only" to avoid inadvertently exporting these in the future? ie.

```purs
-- For internal use only
toNonEmptyString :: String -> NonEmptyString

-- For internal use only
fromNonEmptyString :: NonEmptyString -> String
```

Perhaps it's obvious enough as-is, but I always get a bit wary of modules which manually export a number of functions and intentionally hide a few -- it's easy to accidentally export or to forget to export particular functions.

Reply:
Sounds reasonable to me.